lib/frontend.js

```diff
@@ -86,7 +86,7 @@ var Frontend = function(configuration, smiService){
       // }
 
       // valid client identity
-      if(!from && (config.token && config.token != headers.token)){
+      if(!from || (config.token && config.token !== headers.token)){
```

Comment:
Hey, just come across with this, it seems a interesting use case to extend the libraries with plugins. :)